src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -613,17 +630,38 @@ ApiGlobe.prototype.computeDistance = function computeDistance(p1, p2) {
     return this.scene.getEllipsoid().computeDistance(new GeoCoordinate().copy(p1), new GeoCoordinate().copy(p2));
 };
 
+ApiGlobe.prototype.setSceneLoaded = function computeDistance(callback) {
+    loaded = false;
+    this.callback = () => {
```

Comment:
If you call 2 times `setSceneLoaded` before `globe-loaded` fires, only one of the 2 callbacks would get called. Actually, if you want to execute some piece of code once after the next `globe-loaded` event, promises are a good way to do so. More precisely, you can use what is commonly called **deferred**:

They are built this way (they are just wrappers around promise and their reject and resolve functions).
```js
// in some utils module maybe?
function defer() {
    let defer = {};
    defer.promise = new Promise((resolve, reject) => {
        defer.resolve = resolve;
        defer.reject = reject;
    });
    return defer;
}
```
Then instead of `this.callback` you will instanciate a defer, fire and replace it each time globe-loaded fires:
```js
this.globeLoadedDeferred = defer(); // precise names are better
this.addEventListener('globe-loaded', () => {
    this.globeLoadedDeferred.resolve();
    this.globeLoadedDeferred = defer();
});
```
And each time you currently call `setSceneLoaded`:
```js
this.globeLoadedDeferred.promise.then(<your callback>);
```
This way, each registered  callbacks since last `globe-loaded` would get called the next time the event fires. No need to add and remove listeners.

Also, by returning promises, you can chain them (see below).